File: WorkoutTracker.Domain/Entities/Exercises/ExerciseSets/ExerciseSetGroup.cs
using WorkoutTracker.Domain.Base;
using WorkoutTracker.Domain.Entities.Workouts;

namespace WorkoutTracker.Domain.Entities.Exercises.ExerciseSets;

public class ExerciseSetGroup : IDbModel
{
    public long Id { get; set; }

    public long WorkoutId { get; set; }
    public Workout? Workout { get; set; }

    public long ExerciseId { get; set; }
    public Exercise? Exercise { get; set; }

    public IEnumerable<ExerciseSet> ExerciseSets { get; set; } = null!;
}

File: WorkoutTracker/src/app/shared/models/exercises/exercise-set-group.ts
import { DbModel } from "src/app/shared/models/base/db-model";
import { ExerciseType } from "src/app/exercises/models/exercise-type";
import { ExerciseSet } from "./exercise-set";
import { ModelWeight } from "../model-weight";

export interface ExerciseSetGroup extends DbModel {
    sets: number;
    weight: ModelWeight;

    exerciseId: number;
    exerciseName?: string;
    exerciseType?: ExerciseType;

    exerciseSets: ExerciseSet[];
}